src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2617,6 +2696,51 @@ public static LazyStringValue TableValueToId(JsonOperationContext context, int i
             var ptr = tvr.Read(index, out int size);
             return Slice.From(context.Allocator, ptr, size, ByteStringType.Immutable, out slice);
         }
+
+        [StorageIndexEntryKeyGenerator]
+        private static ByteStringContext.Scope GenerateBucketAndEtagIndexKeyForDocuments(ByteStringContext context, ref TableValueReader tvr, out Slice slice) => 
+            GenerateBucketAndEtagIndexKey(context, idIndex: (int)DocumentsTable.LowerId, etagIndex: (int)DocumentsTable.Etag, ref tvr, out slice);
+
+        [StorageIndexEntryKeyGenerator]
+        private static ByteStringContext.Scope GenerateBucketAndEtagIndexKeyForTombstones(ByteStringContext context, ref TableValueReader tvr, out Slice slice) => 
+            GenerateBucketAndEtagIndexKey(context, idIndex: (int)TombstoneTable.LowerId, etagIndex: (int)TombstoneTable.Etag, ref tvr, out slice);
+
+        internal static ByteStringContext.Scope GenerateBucketAndEtagIndexKey(ByteStringContext context, int idIndex, int etagIndex, ref TableValueReader tvr, out Slice slice)
+        {
+            var idPtr = tvr.Read(idIndex, out var size);
+            var etagPtr = tvr.Read(etagIndex, out _);
+
+            return GenerateBucketAndEtagSlice(context, idPtr, size, etagPtr, out slice);
+        }
+
+        internal static ByteStringContext.Scope ExtractIdFromKeyAndGenerateBucketAndEtagIndexKey(ByteStringContext context, int keyIndex, int etagIndex, ref TableValueReader tvr, out Slice slice)
+        {
+            var keyPtr = tvr.Read(keyIndex, out var keySize);
+
+            int sizeOfDocId = 0;
+            for (; sizeOfDocId < keySize; sizeOfDocId++)
+            {
+                if (keyPtr[sizeOfDocId] == SpecialChars.RecordSeparator)
+                    break;
+            }
+
+            var etagPtr = tvr.Read(etagIndex, out _);
+
+            return GenerateBucketAndEtagSlice(context, keyPtr, sizeOfDocId, etagPtr, out slice);
+        }
+
+        private static ByteStringContext.Scope GenerateBucketAndEtagSlice(ByteStringContext context, byte* idPtr, int idSize, byte* etagPtr, out Slice slice)
```

Comment:
Why are you passing an _etagPtr_ instead of `etag` here?

Reply:
changed to `etag`